psh/test-touch.py

```diff
@@ -29,21 +29,27 @@ def assert_executable(p):
     date = psh.date(p)
     if not ctx.target.rootfs:
         psh.assert_cmd(p, 'touch /syspage/psh', result='success', msg=msg)
-        assert_mtime(p, {'psh': date}, '/syspage')
+        assert_dir_mtimes(p, {'psh': date}, '/syspage')
     else:
         psh.assert_cmd(p, 'touch /bin/psh', result='success', msg=msg)
-        assert_mtime(p, {'psh': date}, '/bin')
+        assert_dir_mtimes(p, {'psh': date}, '/bin')
 
 
 def assert_devices(p):
-    dates = {}
-    devices = psh.ls_simple(p, '/dev')
-    for dev in devices:
-        dates[dev] = psh.date(p)
-        msg = f"Prompt hasn't been seen after the device touch: /dev/{dev}"
-        psh.assert_cmd(p, f'touch /dev/{dev}', result='success', msg=msg)
+    dev_files = psh.ls(p, '/dev')
 
-    assert_mtime(p, dates, dir='/dev')
+    # /dev may include directories - skip them
+    dirs = [file for file in dev_files if file.is_dir]
+    for file in dirs:
+        dev_files.remove(file)
+
+    # check only 5 devices to save time
+    for dev in dev_files[:5]:
```

Comment:
It could be written in a single line, but I'm not sure if that wouldn't reduce readability
it wouldn't make a difference in performance either.
```suggestion
    # /dev may include directories - skip them
    # Get only 5 devices to save time
    dev_files = [file for file in psh.ls(p, '/dev') if not file.is_dir][:5]

    for dev in dev_files:
```

Reply:
Thanks for this idea, I've applied it partly